rust/xaynet-core/src/message/mod.rs

```diff
@@ -29,7 +29,11 @@
 pub(crate) mod message;
 pub(crate) mod payload;
 pub(crate) mod traits;
-pub(crate) mod utils;
+
+// FIXME: I'd like to make this `pub(crate)` but then the doc-tests in
+// utils::chunkable_iterator cannot be compiled
+#[doc(hidden)]
+pub mod utils;
```

Comment:
i'd say it is probably better to mark the code in the doctests as `ignore` and to move their content to some unit tests where we can run them on `pub(crate)` imports. otherwise we would expose internal details, which seems to be a worse tradeoff than some test duplication.

Reply:
I'm not too concerned about test duplication, but the problem is if we make changes to the iterator, we may forget to update the docstring and not notice that it's broken.

Reply:
I did as you suggested in the end.